agent/consul/rpc.go

```diff
@@ -295,7 +295,10 @@ func (s *Server) handleNativeTLS(conn net.Conn) {
 func (s *Server) handleMultiplexV2(conn net.Conn) {
 	defer conn.Close()
 	conf := yamux.DefaultConfig()
-	conf.LogOutput = s.config.LogOutput
+	// override the default because LogOutput conflicts with Logger
+	conf.LogOutput = nil
+	// TODO: should this be created once and cached?
+	conf.Logger = s.logger.StandardLogger(&hclog.StandardLoggerOptions{InferLevels: true})
```

Comment:
Calling out this TODO. I think it would be easy to cache this logger once, instead of having it be created on each new connection. I'm not sure if it is worth it, or is creating a new one is fine each time.